kymatio/scattering1d/core/timefrequency_scattering.py

```diff
@@ -0,0 +1,126 @@
+import math 
+def timefrequency_scattering(x, pad, unpad, backend, J, J_fr, psi1, psi2, phi, 
+                             psi_fr, phi_fr, pad_left=0,pad_right=0, ind_start=None, 
+                             ind_end=None, oversampling=0, size_scattering=(0, 0, 0), 
+                             out_type='array'):
+
+    subsample_fourier = backend.subsample_fourier
+    transpose = backend.transpose
+    modulus = backend.modulus
+    rfft = backend.rfft
+    ifft = backend.ifft
+    irfft = backend.irfft
+    cdgmm = backend.cdgmm
+    concatenate = backend.concatenate
+    to_real = backend.to_real
+    real_out = backend.real_out 
+
+    U_0 = pad(x, pad_left, pad_right)
+    U_0_hat = rfft(U_0)
+
+    # good spot to print shapes
+    U_1_list = []
+    S_1_T_list = []
+    for n1 in range(len(psi1)):
+        j1 = psi1[n1]['j']
+        k1 = max(j1 - oversampling, 0)
+
+        U_1_c = cdgmm(U_0_hat, psi1[n1][0])
+        U_1_hat = subsample_fourier(U_1_c, 2**k1)
+        U_1_c = ifft(U_1_hat)
+
+        U_1_m = modulus(U_1_c)
+        U_1_hat = rfft(U_1_m)
+
+        U_1_list.append(U_1_hat)
+
+        k1_J = max(J - k1 - oversampling, 0)
+
+        #TODO add if statement for T=None, int, 'global'
+        S_1_c = cdgmm(U_1_hat, phi[k1])
+        S_1_hat = subsample_fourier(S_1_c, 2**k1_J)
+        S_1_r = irfft(S_1_hat)
+
+        S_1_T = unpad(S_1_r, ind_start[k1_J + k1], ind_end[k1_J + k1])
+        # good spot to print shapes
+        S_1_T_list.append(S_1_T)
+
+    total_height = 2 ** math.ceil(1+math.log2(len(psi1)))
+    padding_row = 0 * S_1_T
+    for n1 in range(total_height - len(S_1_T_list)):
+        S_1_T_list.append(padding_row)
+    S_1_TM = to_real(concatenate(S_1_T_list))
+
+    k_fr_J = max(J_fr - oversampling, 0)
+    S_1_TM_T = transpose(S_1_TM)
+    S_1_TM_T_hat = rfft(S_1_TM_T)
+    S_1_TM_T_c = cdgmm(S_1_TM_T_hat, phi_fr[0])
```

Comment:
Another, related, worry is about the low pass filter corresponding to frequency, phi_fr. According to the impression I got from [Vincent's code](https://github.com/janden/kymatio/blob/vincent_jtfs/kymatio/scattering1d/core/timefrequency_scattering.py#L76) it seems that there are supposed to be `J_fr` number of phi filters? But there are a smaller number of `phi_fr`'s from what I printed out. Also, the larger the index, the smaller the `phi_fr`